### ISSUE TITLE
bugsnag.asgi.BugsnagMiddleware bug where client scope is None

### ISSUE THREAD

**Issue by hozkok** [bugsnag/bugsnag-python#295]
### Describe the bug
`bugsnag.asgi.BugsnagMiddleware` has a bug in [this line](https://github.com/bugsnag/bugsnag-python/blob/master/bugsnag/asgi.py#L90), causing the following error in fastapi application;

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/bugsnag/middleware.py", line 159, in run
    to_call(event)
  File "/usr/local/lib/python3.8/site-packages/bugsnag/middleware.py", line 65, in __call__
    self.bugsnag(event)
  File "/usr/local/lib/python3.8/site-packages/bugsnag/sessiontracker.py", line 155, in __call__
    self.bugsnag(event)
  File "/usr/local/lib/python3.8/site-packages/bugsnag/middleware.py", line 23, in middleware
    ret = self.before(event)
  File "/usr/local/lib/python3.8/site-packages/bugsnag/asgi.py", line 87, in add_request_info
    if 'client' in scope and len(scope['client']) > 0:
TypeError: object of type 'NoneType' has no len()
```
According to the ASGI spec, [the client scope](https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope) can be `None` object, but checking the object only in `if 'client' in scope` causes an issue if client is None.
> Optional; if missing defaults to None.

### Steps to reproduce
1. Define a fastapi app
2. Wrap the app with `app = BugsnagMiddleware(app)`
3. Raise an exception (and do not handle)
4. See the error

### Environment
* Bugsnag version: 4.0.1
* Python version: 3.8.10
* Integration framework version:
    * ASGI: fastapi==0.66.1

**Comment by luke-belton** [comment]
Hi @hozkok  - are you able to share more about your reproduction case with us please?

We are currently unable to reproduce what you are seeing. I've created a `fastapi` app with the versions above and wrapped it in `BugsnagMiddleware` but unhandled Exceptions are being sent to Bugsnag without issue.

Perhaps you could write into support@bugsnag.com with full code snippets of how you are configuring and initialising Bugsnag for your application?

**Comment by theelous3** [comment]
What is the len of `None` (and is it greater than zero? :D )

 It's a mistake regardless of how easily reproducible a bug this is.

> client (Iterable[Unicode string, int]) – A two-item iterable of [host, port], where host is the remote host’s IPv4 or IPv6 address, and port is the remote port as an integer. Optional; if missing defaults to None.

**Comment by hozkok** [comment]
Hi @luke-belton,

I was able to reproduce the error using raw TCP socket on uvicorn server. If the server runs with TCP port specified on `uvicorn`, bugsnag sends the error and doesn't raise an error (as scope client will be bound to that socket).

However, if the server runs with the following config;

```py
    uvicorn.run(
        app,
        loop="uvloop",
        uds="some.sock",
        log_level="debug",
    )
```

Then you will be able to see the bug.

**Comment by luke-belton** [comment]
Thanks @hozkok, we're going to look into this - we'll keep you updated here with any changes.

**Comment by xander-jones** [comment]
Hey @hozkok, @theelous3 👋. A fix for this has been implemented in [v4.1.1](https://github.com/bugsnag/bugsnag-python/releases/tag/v4.1.1), thanks for raising the issue with us :)
